pdl_syntax.md

```diff
@@ -0,0 +1,843 @@
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+fixed MD5 16
+```
+In the above example, `16` is the defined size of bytes for MD5 schema.
+
+
+## Typerefs
+Pegasus supports a new schema type known as a typeref. A typeref is like
+a typedef in C. It does not declare a new type but declares an alias to
+an existing type.
+
+### Typerefs can be used to name anonymous types.
+
+It is very useful, because unions, maps and arrays cannot be named directly like records and enums.
+
+For example:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+typeref AnswerTypes = union[MultipleChoice, TextEntry]
+
+```
+
+Typerefs can be referred to from any other type using the name.
+
+For example:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Question {
+  answerFormat: AnswerTypes
+}
+```
+      
+
+### Typerefs can provide additional clarity when using primitive types.
+
+For example:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+typeref UnixTimestamp = long
+```
+
+### Typerefs can be used to specify custom types and coercers
+For example, Joda time has a convenient DateTime class. If we wish to use this class in Java to represent date times, all we need to do is define a pegasus custom type that binds to it:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+@java.class = "org.joda.time.DateTime"
+@java.coercerClass = "com.linkedin.example.DateTimeCoercer"
+typeref DateTime = string
+```
+The coercer is responsible for converting the pegasus “referenced” type, in this case "string" to the Joda DateTime class:
+
+Once a custom type is defined, it can be used in any type. 
+
+For example, to use the DateTime custom type in a record:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Fortune {
+  createdAt: DateTime
+} 
+```
+
+## Namespace
+
+Namespace is used to qualify the namespace for the named schema.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Foo {}
+```
+
+## Import
+
+Imports are optional statements which allow you to avoid wirting the full qualified names, similar to in Java.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.test.idl.imports
+
+import com.linkedin.pegasus.generator.test.idl.records.Simple
+
+record Example {
+  /**
+   * Requires an import since this type is outside the root namespace and is not declared in this file.
+   */
+  externalOutsideNS: Simple
+}
+```
+
+**Note:** 
+- Any type that is not imported and is not within the namespace from which it's referenced must be referenced by
+fully qualified name.
+- Using imports in the following ways will lead to PDL parser errors. You should avoid to do so.
+1. Importing types declared inside the document.
+2. Importing types within the root namespace of the document.
+3. Declaring types that conflict with existing imports.
+
+## Deprecation
+
+All types, enum symbols and record fields can be deprecated by adding @deprecated annotation.
```

Comment:
In general in the documentation, escape code-like text such as "@deprecated" with the backtick so that it shows up like `@deprecated`

Reply:
Updated.